### DISCUSSION TITLE
contributing guide

### DISCUSSION THREAD

**Discussion by danish-mehmood** [rayepps/radash#16] [Ideas]
is this project open to contributions , if yes then we should come up with a contribution guide and also if there is any road man for the future features than the maintainer should let the potential contributor know of that as well

**Comment by sodiray** [comment]
Hey, thanks for bringing this up. I'll add a contributions guide by the end of the weekend. I'm also planning to revamp the README to give developers a bit more helpful information as well.

I know there are a lot of contribution guides and ethos out there. Since you brought it up, do you have an open source repo you think does it well that I could reference?

**Reply by danish-mehmood** [reply]
i can not think of specific contribution guide which i could reference but i have some suggestions as a user of open source , following should be part of any contribution guide 
- rules to follow (code conventions e.t.c)
- step by step guide to add a new feature or resolve an issue (this is something missing from most of the guides the writers often think that everyone who is going to contribute to the project is an expert at open source when that is not the case , guide should always assume that the person contributing is a first timer and has never before contributed to the open source)
- explanation of tag system the project follows 
- branch naming conventions
- where to find help if stuck 
- how to write the tests for added features 
these are the points i can think of right now , if i find a good reference i will definitely let you know , but the fore mentioned points are of utmost importance

**Comment by sodiray** [comment]
Here's [a draft](https://github.com/rayepps/radash/pull/29)
